cpp/tests/io/json_whitespace_normalization_test.cu

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <io/fst/lookup_tables.cuh>
+#include <io/utilities/hostdevice_vector.hpp>
+
+#include <cudf/scalar/scalar_factories.hpp>
+#include <cudf/types.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/cudf_gtest.hpp>
+#include <cudf_test/testing_main.hpp>
+
+#include <rmm/cuda_stream.hpp>
+#include <rmm/cuda_stream_view.hpp>
+
+#include <thrust/iterator/discard_iterator.h>
+
+#include <cstdlib>
+#include <string>
+
+namespace {
+// Type used to represent the atomic symbol type used within the finite-state machine
+using SymbolT = char;
+using StateT  = char;
+
+// Type sufficiently large to index symbols within the input and output (may be unsigned)
+using SymbolOffsetT = uint32_t;
+
+enum class dfa_states : char { TT_OOS = 0U, TT_DQS, TT_DEC, TT_WS, TT_NUM_STATES };
+enum class dfa_symbol_group_id : uint32_t {
+  DOUBLE_QUOTE_CHAR,  ///< Quote character SG: "
+  ESCAPE_CHAR,        ///< Escape character SG: '\\'
+  NEWLINE_CHAR,       ///< Newline character SG: '\n'
+  WHITESPACE_CHAR,    ///< Whitespace character SG: ' '
+  TABSPACE_CHAR,      ///< Tabspace character SG: '\t'
```

Comment:
Sorry, I just recognized that me calling the common symbol group for both space and tab `WHITESPACE_CHAR` was a poor choice, since it sounds like the group wouldn't comprise the tab. We probably want to find a better name; maybe just `WHITESPACE`?

Reply:
How about calling it `WHITESPACE_SYMBOLS` or `WHITESPACE_CHARS`?

Reply:
Sounds good. Either one works for me